pkg/krmtotf/resource.go

```diff
@@ -322,6 +326,21 @@ func (r *Resource) AllTopLevelFieldsAreImmutableOrComputed() bool {
 	return true
 }
 
+func getObservedStateFromStatus(status map[string]interface{}) map[string]interface{} {
+	observedState, exists, _ := unstructured.NestedMap(status, k8s.ObservedStateFieldName)
+	if !exists {
+		observedState = make(map[string]interface{})
```

Comment:
Probably safer to just return nil, lest someone try to write to it

Reply:
#1430